include/libopencm3/stm32/l4/syscfg.h

```diff
@@ -0,0 +1,122 @@
+/** @defgroup syscfg_defines SYSCFG Defines
+ *
+ * @ingroup STM32L4xx_defines
+ *
+ * @brief Defined Constants and Types for the STM32L4xx Sysconfig
+ *
+ * @version 1.0.0
+ *
+ * @date 28 December 2017
+ *
+ * LGPL License Terms @ref lgpl_license
+ */
+
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * Copyright (C) Bruno Randolf <br1@einfach.org>
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef LIBOPENCM3_SYSCFG_H
+#define LIBOPENCM3_SYSCFG_H
+/**@{*/
+
+/* --- SYSCFG registers ---------------------------------------------------- */
+
+#define SYSCFG_MEMRMP			MMIO32(SYSCFG_BASE + 0x00)
+#define SYSCFG_CFGR1			MMIO32(SYSCFG_BASE + 0x04)
+#define SYSCFG_EXTICR(i)		MMIO32(SYSCFG_BASE + 0x08 + (i)*4)
+#define SYSCFG_EXTICR1			SYSCFG_EXTICR(0)
+#define SYSCFG_EXTICR2			SYSCFG_EXTICR(1)
+#define SYSCFG_EXTICR3			SYSCFG_EXTICR(2)
+#define SYSCFG_EXTICR4			SYSCFG_EXTICR(3)
+#define SYSCFG_SCSR			MMIO32(SYSCFG_BASE + 0x18)
+#define SYSCFG_CFGR2			MMIO32(SYSCFG_BASE + 0x1C)
+#define SYSCFG_SWPR			MMIO32(SYSCFG_BASE + 0x20)
+#define SYSCFG_SKR			MMIO32(SYSCFG_BASE + 0x24)
+#define SYSCFG_SWPR2			MMIO32(SYSCFG_BASE + 0x29)
+
+/* --- SYSCFG_MEMRMP Values ------------------------------------------------ */
+
+#define SYSCFG_CFGR1_MEM_MODE_MASK	7
+#define SYSCFG_CFGR1_MEM_MODE_FLASH	0
+#define SYSCFG_CFGR1_MEM_MODE_SYSTEM	1
+#define SYSCFG_CFGR1_MEM_MODE_FMC	2
+#define SYSCFG_CFGR1_MEM_MODE_SRAM	3
+#define SYSCFG_CFGR1_MEM_MODE_QSPI	6
+
+/* --- SYSCFG_CFGR1 Values ------------------------------------------------- */
+
+#define SYSCFG_CFGR2_FPU_IE5		(1 << 31)
+#define SYSCFG_CFGR2_FPU_IE4		(1 << 30)
+#define SYSCFG_CFGR2_FPU_IE3		(1 << 29)
+#define SYSCFG_CFGR2_FPU_IE2		(1 << 28)
+#define SYSCFG_CFGR2_FPU_IE1		(1 << 27)
+#define SYSCFG_CFGR2_FPU_IE0		(1 << 26)
```

Comment:
really not sure this is a useful way of defining them. (the fpu bits here I mean)

Reply:
i made a mistake: this is  SYSCFG_CFGR1_* anyhow. will update the PR.

As for the FPU bits, should I try to find short names according to
FPU_IE[5]: Inexact interrupt enable
FPU_IE[4]: Input denormal interrupt enable
FPU_IE[3]: Overflow interrupt enable
FPU_IE[2]: underflow interrupt enable
FPU_IE[1]: Divide-by-zero interrupt enable
FPU_IE[0]: Invalid operation interrupt enable
?

Reply:
Maybe even just leave them out for now.  Perhaps just list their size and that they exist? Sometimes it's best to get the file started, without trying to get everything in.

Reply:
Well, I find naming them like I did in my last update makes most sense:
```
#define SYSCFG_CFGR1_FPU_IE_INEXACT	(1 << 31)
#define SYSCFG_CFGR1_FPU_IE_DENORMAL	(1 << 30)
#define SYSCFG_CFGR1_FPU_IE_OVERFLOW	(1 << 29)
#define SYSCFG_CFGR1_FPU_IE_UNDERFLOW	(1 << 28)
#define SYSCFG_CFGR1_FPU_IE_DIVZERO	(1 << 27)
#define SYSCFG_CFGR1_FPU_IE_INVALID	(1 << 26)
```

Or... if you prefer, we could only do
```
/* FPU interrupt enable (x from 0 to 5)
#define SYSCFG_CFGR1_FPU_IE(x)		(1 << (26 + x))
```

In any case, for me either way is fine...